### PR TITLE
[SYCLomatic] Enable migration of  CUBLASLT_EPILOGUE_GELU & GELU_AUX

### PR DESCRIPTION
**Pull request by abhilash1910**
cc @zhiweij1 @zhimingwang36

### CODE REVIEW

File: clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp [hunk 1]
```diff
@@ -999,8 +1000,14 @@ inline sycl::event matmul(descriptor_ptr handle, matmul_desc_ptr compute_desc,
...
 
   if (compute_desc->_epilogue != epilogue_t::nop) {
+    
     ::dnnl::post_ops matmul_ops;
+    if(compute_desc->_epilogue == epilogue_t::relu){
     matmul_ops.append_eltwise(::dnnl::algorithm::eltwise_relu, 0.f, 0.f);
+    }
+    else if(compute_desc->_epilogue == epilogue_t::gelu){
+    matmul_ops.append_eltwise(::dnnl::algorithm::eltwise_gelu_erf, 0.f, 0.f);
+    }
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp]: Format the code
```suggestion
  if (compute_desc->_epilogue != epilogue_t::nop) {
    ::dnnl::post_ops matmul_ops;
    if (compute_desc->_epilogue == epilogue_t::relu) {
      matmul_ops.append_eltwise(::dnnl::algorithm::eltwise_relu, 0.f, 0.f);
    } else if (compute_desc->_epilogue == epilogue_t::gelu) {
      matmul_ops.append_eltwise(::dnnl::algorithm::eltwise_gelu_erf, 0.f, 0.f);
    }
```

File: clang/test/dpct/cublaslt.cu [hunk 2]
```diff
@@ -220,9 +220,13 @@ void foo3() {
   // CHECK: dpct::blas_gemm::experimental::epilogue_t e;
   // CHECK-NEXT: e = dpct::blas_gemm::experimental::epilogue_t::nop;
   // CHECK-NEXT: e = dpct::blas_gemm::experimental::epilogue_t::relu;
+  // CHECK-NEXT: e = dpct::blas_gemm::experimental::epilogue_t::gelu;
+  // CHECK-NEXT: e = dpct::blas_gemm::experimental::epilogue_t::gelu;
   cublasLtEpilogue_t e;
   e = CUBLASLT_EPILOGUE_DEFAULT;
   e = CUBLASLT_EPILOGUE_RELU;
+  e = CUBLASLT_EPILOGUE_GELU;
+  e = CUBLASLT_EPILOGUE_GELU_AUX;
```
Comment (zhiweij1) [inline clang/test/dpct/cublaslt.cu:241]: GELU_AUX is not as same as GELU.
According to CUBLAS doc, GELU_AUX  is
```
Apply GELU point-wise transform to the results (x := GELU(x)). This epilogue mode outputs GELU input as a separate matrix (useful for training). See CUBLASLT_MATMUL_DESC_EPILOGUE_AUX_POINTER of [cublasLtMatmulDescAttributes_t](https://docs.nvidia.com/cuda/cublas/index.html#cublasltmatmuldescattributes-t).
```
So you need update the implementation to copy the GELU input to EPILOGUE_AUX_POINTER

File: clang/test/dpct/cublaslt.cu [hunk 3]
```diff
@@ -220,9 +220,13 @@ void foo3() {
   // CHECK: dpct::blas_gemm::experimental::epilogue_t e;
   // CHECK-NEXT: e = dpct::blas_gemm::experimental::epilogue_t::nop;
   // CHECK-NEXT: e = dpct::blas_gemm::experimental::epilogue_t::relu;
+  // CHECK-NEXT: e = dpct::blas_gemm::experimental::epilogue_t::gelu;
+  // CHECK-NEXT: e = dpct::blas_gemm::experimental::epilogue_t::gelu;
   cublasLtEpilogue_t e;
   e = CUBLASLT_EPILOGUE_DEFAULT;
   e = CUBLASLT_EPILOGUE_RELU;
+  e = CUBLASLT_EPILOGUE_GELU;
```
Comment (zhiweij1) [inline clang/test/dpct/cublaslt.cu:240]: Please also add E2E tests for those 2 epilogues in: https://github.com/oneapi-src/SYCLomatic-test/blob/SYCLomatic/features/feature_case/cublasLt/matmul.cu

**Review by zhiweij1 — CHANGES_REQUESTED** [review]
Please update the PR.

File: clang/test/dpct/cublaslt.cu (hunk 3 above)
Comment (abhilash1910) [reply clang/test/dpct/cublaslt.cu:240]: OK

**Comment by abhilash1910** [comment]
Refer to #2460 . Closing this.

**Pull request closed without merging**